crates/accelerate/src/sparse_observable/mod.rs

```diff
@@ -3216,6 +3464,34 @@ impl PySparseObservable {
         Ok(simplified.into())
     }
 
+    /// Greedily combine the terms in the observable.
```

Comment:
Maybe we should say something about runtime here. Did you find this to be slow in practice?

Reply:
I added a comment about runtime: the worst-time complexity is $O(terms^3 * qubits)$, however in practice it's significantly lower: on HamLib benchmarks we never have more than 2-3 total iterations. In addition, due to fast exit conditions (for trying to combine two sparse terms) + sparsity of the indices, we usually do a small number of computations to decide that two sparse terms cannot be merged or merge them. Thus I would speculate that in practice we see $O(terms^2)$ runtime. The algorithm is indeed fast on HamLib benchmarks. If the runtime does become a bottleneck, we could limit the total number of iterations by a small constant.